api-reference/user-app/vitals/vitals.yaml

```diff
@@ -0,0 +1,203 @@
+openapi: 3.1.0
+info:
+  title: Vitals API Contract
+  description: Documentation for the Vitals API endpoint.
+  version: 1.0.0
+  license:
+    name: Proprietary License
+    url: https://example.com/license
+
+servers:
+  - url: https://api.eka.care
+    description: Production server
+  - url: https://api.dev.eka.care
+    description: Development server
+
+tags:
+  - name: Vitals
+    description: API related to managing patient's vitals.
+
+paths:
+  /api/v1/vitals/{txn_id}:
+    put:
+      tags:
+        - Vitals
+      summary: Update Patient Vitals
+      description: Update the vitals data for a specific patient using the transaction ID.
+      parameters:
+        - name: txn_id
+          in: path
+          required: true
+          schema:
+            type: string
+          description: Transaction ID to identify the specific vitals record.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                vitals:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/Vital'
+              example:
+                vitals:
+                  - vital_display_name: heart_rate
+                    vital_loinc_code: 8867-4
+                    vital_value: 
+                      vital_numeric_value: 75
+                      vital_boolean_value: null
+                      vital_string_value: null
+                    unit_display_name: '{Counts}/min'
+                    unit_ucum_code: '{Counts}/min'
+                    measured_at:
+                      effectiveDateTime: '1693802400'
```

Comment:
take this as a string and allow different formats like `year` or `year+month` or `year+month+date ...` etc , check FHIR once for this